base/dcp_client.go

```diff
@@ -116,24 +116,26 @@ func NewDCPClient(ID string, callback sgbucket.FeedEventCallbackFunc, options DC
 }
 
 func (dc *DCPClient) Start() (doneChan chan error, err error) {
-
 	err = dc.initAgent(dc.spec)
 	if err != nil {
-		return nil, err
+		return dc.doneChannel, err
 	}
 	dc.startWorkers()
 
 	for i := uint16(0); i < dc.numVbuckets; i++ {
 		openErr := dc.openStream(i)
 		if openErr != nil {
-			return nil, fmt.Errorf("Unable to start DCP client, error opening stream for vb %d: %w", i, openErr)
+			// If this error condition is hit, the caller must call DCPClient.Close(). If we close here, should we make it blocking?
```

Comment:
Can you add a comment for Start() to clarify that this is the contract?  (that callers must always call close on DCPClient, even in error cases).